docs/new-architecture-app-renderer-android.md

```diff
@@ -63,11 +68,65 @@ public class MyApplication extends Application implements ReactApplication {
 }
 ```
 
+</TabItem>
+
+<TabItem value="kotlin">
+
+```kotlin title='MyApplication.kt'
+
+class MyApplication() : Application(), ReactApplication {
+  private val mReactNativeHost: ReactNativeHost =
+      object : ReactNativeHost(this) {
+        // Add those lines:
+        @get:Nullable
+        protected val jSIModulePackage: JSIModulePackage
+          protected get() =
+              object : JSIModulePackage() {
+                fun getJSIModules(
+                    reactApplicationContext: ReactApplicationContext?,
+                    jsContext: JavaScriptContextHolder?
+                ): List<JSIModuleSpec> {
+                  val specs: MutableList<JSIModuleSpec> = ArrayList()
+                  specs.add(
+                      object : JSIModuleSpec() {
+                        val jSIModuleType: JSIModuleType
+                          get() = JSIModuleType.UIManager
+                        val jSIModuleProvider: JSIModuleProvider<UIManager>
+                          get() {
+                            val componentFactory: ComponentFactory = ComponentFactory()
+                            CoreComponentsRegistry.register(componentFactory)
+                            val reactInstanceManager: ReactInstanceManager =
+                                getReactInstanceManager()
+                            val viewManagerRegistry: ViewManagerRegistry =
+                                ViewManagerRegistry(
+                                    reactInstanceManager.getOrCreateViewManagers(
+                                        reactApplicationContext))
+                            return FabricJSIModuleProvider(
+                                reactApplicationContext,
+                                componentFactory,
+                                EmptyReactNativeConfig(),
+                                viewManagerRegistry)
+                          }
+                      })
```

Comment:
I'd suggest you take a look at this class:
https://github.com/facebook/react-native/blob/main/ReactAndroid/src/main/java/com/facebook/react/defaults/DefaultJSIModulePackage.kt

As it's similar to what you're doing here and it's a bit more idiomatic 👍 Feel free to ask questions if you have any.